### PR TITLE
dhcp: about rapid commit

### PR DESCRIPTION
**Pull request by yuwata**
One fix for client side, and add server side support.

### CODE REVIEW

File: man/systemd.network.xml [hunk 1]
```diff
@@ -3768,6 +3768,18 @@ ServerAddress=192.168.0.1/24</programlisting>
         </listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>RapidCommit=</varname></term>
+        <listitem>
+          <para>Takes a boolean. When true, the server supports
+          <ulink url="https://datatracker.ietf.org/doc/html/rfc4039">RFC 4039</ulink>. When a client send
```
Comment (keszybz) [inline man/systemd.network.xml]: "client sends a DHCPDISCOVER message"
"then the server will reply with a DHCPACK message"

**Review by keszybz — COMMENTED** [review]
Looks nice and clean. Do you intend this for v255?

**Comment by keszybz** [comment]
`test-dhcp-server` fails with SIGABRT.

**Comment by yuwata** [comment]
> Looks nice and clean. Do you intend this for v255?

Yes. I'd like so.

**Comment by keszybz** [comment]
Ack. LGTM.

File: src/libsystemd-network/sd-dhcp-client.c [hunk 2]
```diff
@@ -1367,16 +1367,15 @@ static int client_timeout_resend(
...
 
+                client->request_sent = time_now;
                 break;
 
         case DHCP_STATE_SELECTING:
                 r = client_send_discover(client);
                 if (r < 0 && client->attempt >= client->max_attempts)
                         goto error;
 
-                if (client->rapid_commit)
```
Comment (rpigott) [inline src/libsystemd-network/sd-dhcp-client.c:1377]: Why is this unnecessary? If client->rapid_commit is false, we didn't set the rapid_commit option in the DISCOVER query, right? Does this handle the case of invalid ACK response from the server?

Comment (yuwata) [reply src/libsystemd-network/sd-dhcp-client.c:1377]: Actually the value is not used, at least currently, when rapid commit is not enabled. But, setting this does not hurt anything. Just for simplicity and safety.

**Pull request merged**